src/test/java/hw2/FirstExercise.java

```diff
@@ -0,0 +1,94 @@
+
+        //5. Menu buttons are displayed and have proper texts
+        List<String> expectedMenuItemsText = Arrays.asList("HOME", "CONTACT FORM", "SERVICE", "METALS & COLORS");
+        List<WebElement> actualMenuItems = driver.findElements(By.cssSelector(".m-l8 > li"));
+
+        checkElementsAreDisplayed(actualMenuItems);
+
+        List<String> actualMenuItemsText = getMenuItemsText(actualMenuItems);
+        asrt.assertEquals(actualMenuItemsText, expectedMenuItemsText);
+
+        //6. Images are displayed
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.className("benefit-icon")));
+        List<WebElement> icons = driver.findElements(By.className("benefit-icon"));
+        checkElementsAreDisplayed(icons);
+
+        //7. Texts are displayed and equal to expected
+        List<String> expectedIconTexts = Arrays.asList(
+                "To include good practices\n" +
+                        "and ideas from successful\n" +
+                        "EPAM project",
+                "To be flexible and\n" +
+                        "customizable",
+                "To be multiplatform",
+                "Already have good base\n" +
+                        "(about 20 internal and\n" +
+                        "some external projects),\n" +
+                        "wish to get more…"
+        );
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.className("benefit-txt")));
+        List<WebElement> iconTextsElements = driver.findElements(By.className("benefit-txt"));
+        checkElementsAreDisplayed(iconTextsElements);
+
+        List<String> actualIconTexts = getMenuItemsText(iconTextsElements);
+        asrt.assertEquals(actualIconTexts, expectedIconTexts);
+
+        //8. The iframe exists
+        WebElement iframe = wait.until(ExpectedConditions.visibilityOfElementLocated((By.id("frame"))));
+        asrt.assertTrue(iframe.isDisplayed());
+
+        //9. The “Frame Button” exists
+        driver.switchTo().frame("frame");
+        WebElement iframeButton = wait.until(ExpectedConditions.visibilityOfElementLocated((By.id("frame-button"))));
+        asrt.assertTrue(iframeButton.isDisplayed());
+
+        //10. Driver has focus on the original window
+        driver.switchTo().defaultContent();
+
+        //11. Left section menu items are displayed and have proper text
+        List<WebElement> leftSideMenuItems = driver.findElements(By.cssSelector(".sidebar-menu > li"));
+        checkElementsAreDisplayed(leftSideMenuItems);
+        List<String> actualLeftSideMenuNames = getMenuItemsText(leftSideMenuItems);
+        List<String> expectedLeftSideMenuNames = Arrays.asList(
+                "Home",
+                "Contact form",
+                "Service",
+                "Metals & Colors",
+                "Elements packs"
+        );
+        asrt.assertEquals(actualLeftSideMenuNames, expectedLeftSideMenuNames);
+
```

Comment:
How we will know that our assertions passed?

Reply:
We can add assertAll() to the end of the test, than the test will fail if one of the assertions inside fails.